### PR TITLE
bugfix: unicode attribute values and variable values (#155)

### LINKED ISSUE #155: problems with unicode attribute values and variable values
When using a unicode string as attribute value or variable value, a UnicodeEncodeError is thrown in
pydap/lib.py", line 135, in encode:
    return '"{0}"'.format(obj)

### PR DESCRIPTION
**Pull request by pauleckhardt**
Fixes #155

### CODE REVIEW

File: src/pydap/lib.py [hunk 1]
```diff
@@ -120,14 +119,30 @@ def quote(name):
...
+
+    """
+    safe = ''.join([chr(x) for x in range(128)])
+    return quote_(name.encode('utf-8'), safe=safe)
 
 def encode(obj):
     """Return an object encoded to its DAP representation."""
+    try:
+        return quote_val(obj)
+    except:
```
Comment (laliberte) [inline src/pydap/lib.py]: Which exception is expected here? You should probably except only that one here.

File: src/pydap/responses/dods.py [hunk 2]
```diff
@@ -203,18 +204,26 @@ def _basetype(var):
...
-                        raise TypeError("Could not convert word '{0}' to bytes"
-                                        .format(word))
+                    try:
+                        enc = quote_val(word)
+                        length = len(enc)
+                        yield tostring_with_byteorder(
+                                    np.array(length),
+                                    np.dtype(DAP2_ARRAY_LENGTH_NUMPY_TYPE))
+                        yield enc
+                    except:
```
Comment (laliberte) [inline src/pydap/responses/dods.py]: Again here, what is your try clause capturing? It is too wide and "yield tostring_wit..." is repeated. This must be avoided.

File: src/pydap/lib.py (hunk 1 above)
Comment (tomkralidis) [reply src/pydap/lib.py]: Echo'ing @laliberte, @pauleckhardt can we be more explicit here?

File: src/pydap/responses/dods.py (hunk 2 above)
Comment (tomkralidis) [reply src/pydap/responses/dods.py]: Echo @laliberte, need to be more explicit here.

**Comment by captainkirk99** [comment]
This is a very old PR. Is it still active?

**Comment by Mikejmnez** [comment]
Is this PR active?

**Comment by Mikejmnez** [comment]
Closing this issue as it is very old

**Pull request closed without merging**